### PR TITLE
Remove ListProof class as redundant [ECR-3429]

### PR DESCRIPTION
**Pull request by MakarovS**
## Overview
Remove ListProof class as redundant and replace it with UncheckedListProofAdapter.

---
See: https://jira.bf.local/browse/ECR-3429

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [x] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [x] Public API has Javadoc
- [x] Method preconditions are checked and documented in the Javadoc of the method
- [x] Changelog is updated if needed (in case of notable or breaking changes)
- [x] The continuous integration build passes

### CODE REVIEW

**Comment by skletsun** [comment]
Should it be mentioned in the Changelog?

File: exonum-java-binding/common/src/main/java/com/exonum/binding/common/proofs/list/UncheckedListProof.java [hunk 1]
```diff
@@ -30,5 +30,5 @@
   /**
    * Returns raw source proof of this UncheckedListProof.
```
Comment (bullet-tooth) [inline exonum-java-binding/common/src/main/java/com/exonum/binding/common/proofs/list/UncheckedListProof.java]: shouldn't the JavaDoc be updated?

File: exonum-java-binding/core/src/test/java/com/exonum/binding/core/storage/indices/ProofListContainsMatcher.java [hunk 2]
```diff
@@ -18,15 +18,18 @@
...
 
+import com.exonum.binding.common.collect.MapEntry;
 import com.exonum.binding.common.proofs.list.CheckedListProof;
 import com.exonum.binding.common.proofs.list.UncheckedListProof;
-import java.util.Collections;
-import java.util.List;
-import java.util.Map;
-import java.util.TreeMap;
+
+import java.util.*;
```
Comment (bullet-tooth) [inline exonum-java-binding/core/src/test/java/com/exonum/binding/core/storage/indices/ProofListContainsMatcher.java]: 😱

Comment (MakarovS) [reply exonum-java-binding/core/src/test/java/com/exonum/binding/core/storage/indices/ProofListContainsMatcher.java]: Sorry, my IDEA annoyingly does that, couldn't get myself to reconfigure that. Just did though! Thanks.

File: exonum-java-binding/common/src/main/java/com/exonum/binding/common/proofs/list/UncheckedListProof.java (hunk 1 above)
Comment (MakarovS) [reply exonum-java-binding/common/src/main/java/com/exonum/binding/common/proofs/list/UncheckedListProof.java]: Yes, thanks!

**Comment by MakarovS** [comment]
> Should it be mentioned in the Changelog?

The removed class is an internal one and was not exposed in our API, so not sure. I've put it there just in case.

**Comment by vitvakatu** [comment]
> The removed class is an internal one and was not exposed in our API, so not sure. I've put it there just in case.

Changelog is for our users, so mentioning internal things might be an overkill.

**Comment by skletsun** [comment]
> > The removed class is an internal one and was not exposed in our API, so not sure. I've put it there just in case.
> 
> Changelog is for our users, so mentioning internal things might be an overkill.

Well, then it looks a little redundant, please remove 🙃

**Comment by MakarovS** [comment]
> Well, then it looks a little redundant, please remove 🙃

Removed corresponding changelog update.

**Pull request merged**